### ISSUE TITLE
quickopen is terribly slow on large workspaces

### ISSUE THREAD

**Issue by dibalavs** [microsoft/vscode#79860]
Hello.
I often use QuickOpen feature (CTRL + P) but it works very very very slow first time on large project (just arount 20-30 seconds). 
After first time result was found, second times search will fast (for some time).  If vscode was restarted, or after number of hours, searching the same file will be slow again.

My workspace contains :
- whole sources of QT components (qtbase, qtdeclarative, qtwebengine ...)
- whole sources of chromium project (part of qtwebengine)
-  and some other little libraries.
Some statistics:
*.h files: 56510
*.c* files: 117608
*.mk files: 812
*.txt files: 10041

Workspace is placed on fast PCI-e NVM storage (512 GB). 
My computer is Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz with 24GB RAM.

- VSCode Version: Version: 1.37.1 Commit: f06011ac164ae4dc8e753a3fe7f9549844d15e35
- OS Version: Ubuntu 18.04.3 LTS

Steps to Reproduce:
1.  create workspace with sources of all qt libraries and sources of chromium project
2. do usual work and sometimes search file by pressing CTRL+p 

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
During the 20-30 seconds while the search is running, please open the process explorer (under help menu) and check which process is using CPU. Probably you will see an rg process using high CPU and disk. If so there probably isn't anything that can improve it, besides excluding folders or opening a smaller portion of the project. But if a vscode process is busy, then maybe there is a bug.

**Comment by dibalavs** [comment]
Hello. 
Looks like search is fails when I try to edit search request duringt typing (pressing "backspace" key)
I do screenshot on freeze and record gif animation of search request.

On screenshot:
- right monitor: 
    at the top - search request, 
    at the bottom - right: internal process explorer window.

- left monitor:
   at the right: output of "top" utility
   at the left: output of "iotop" utility
   at the bottom left: output of "fatrace" utility 

![Screenshot from 2019-09-02 13-24-20](https://user-images.githubusercontent.com/801951/64109725-f1f32180-cd88-11e9-9d07-40445e993a7c.png)

gif animation is about attempt to search "accounting-allocator.cc" file.  I write it with errors, press "backspace" and correct errors. 2 times search was ok, but on 3-rd time it "freeze"

![Peek 2019-09-02 13-35](https://user-images.githubusercontent.com/801951/64109740-fb7c8980-cd88-11e9-94df-76a5a02b682b.gif)

**Comment by roblourens** [comment]
Thanks, it looks like `top` disagrees with our process monitor, do you think ours stopped updating here? If top is accurate I would like to see the full command of the processes using high CPU.

I'm not sure how to interpret the fatrace results, do you think it shows anything interesting?

**Comment by dibalavs** [comment]
>  it looks like top disagrees with our process monitor, do you think ours stopped updating here?
No, embedded process monitor is working, numbers are updating, but do not match with 'top' tool.
I right now checked, top shows 100% of cpu usage., but process monitor - 12%.
I think it is related to measurement calculation. 
In case of top - 100% means: 1 CPU eats 100% time.
In case of embedded process explorer - 100% means: 8 cpus eat 100% time. 
But I am not sure, This is my assumption.

>I'm not sure how to interpret the fatrace results, do you think it shows anything interesting?
Fatrace shows file access in real-time. So in case of search it shows:
1) during typing there are a lot of file accesses (rg tool scans whole workspace)
2) when "freeze" is occurred, and "code" process consumes cpu time, there are no file access of rg tool, only from "cpuUsage.sh" used by embedded process explorer.
 

> If top is accurate I would like to see the full command of the processes using high CPU.
they are:
I updated vscode to version 1.38 and cannot reproduce this issue right now. I will  check during some time. If this but is reproduced, I will update this ticket.

**Comment by roblourens** [comment]
Thanks, I also fixed an issue that could possibly be related, this is in Insiders. So you could try that too if you still see it in 1.38.
